source/Cosmos.System2/FileSystem/ManagedPartition.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Cosmos.HAL.BlockDevice;
+
+namespace Cosmos.System.FileSystem
+{
+    public class ManagedPartition
+    {
+        public readonly Partition Host;
+        /// <summary>
+        /// The root path of the file system. Example: 0:\
+        /// </summary>
+        public string RootPath = "";
+        /// <summary>
+        /// The FileSystem object. Null if not mounted.
+        /// </summary>
+        public FileSystem MountedFS;
+        /// <summary>
+        /// Does the partition have a known file system?
+        /// </summary>
+        public bool HasFileSystem
+        {
+            get
+            {
+                foreach (var item in Disk.RegisteredFileSystems)
+                {
+                    if (item.IsType(Host))
+                    {
+                        return true;
+                    }
+                }
+                return false;
+            }
+        }
+
+
+        public ManagedPartition(Partition host)
+        {
+            Host = host;
+        }
+        /// <summary>
+        /// Mounts the partition
+        /// </summary>
+        public void Mount()
```

Comment:
Maybe also add a version of the Mount function where you pass the FilesystemFactory in as well